mediator/esmFldsExchange_hafs_mod.F90

```diff
@@ -168,249 +143,105 @@ subroutine esmFldsExchange_hafs_advt(gcomp, phase, rc)
-    ! ---------------------------------------------------------------------
-    call addfld(fldListFr(compatm)%flds , 'Faxa_sen')
-    call addfld(fldListTo(compocn)%flds , 'Foxx_sen')
-
-    ! ---------------------------------------------------------------------
-    ! to ocn: surface latent heat flux and evaporation water flux
-    ! ---------------------------------------------------------------------
-    call addfld(fldListFr(compatm)%flds , 'Faxa_lat')
-    call addfld(fldListTo(compocn)%flds , 'Foxx_lat')
-
     ! ---------------------------------------------------------------------
-    ! to ocn: sea level pressure from atm
-    ! to ocn: zonal wind at the lowest model level from atm
-    ! to ocn: meridional wind at the lowest model level from atm
-    ! to ocn: wind speed at the lowest model level from atm
-    ! to ocn: temperature at the lowest model level from atm
-    ! to ocn: sea surface skin temperature
-    ! to ocn: specific humidity at the lowest model level from atm
+    ! to atm: surface temperatures from ocn
     ! ---------------------------------------------------------------------
-    allocate(flds(7))
-    flds = (/'Sa_pslv', &
-             'Sa_u   ', &
-             'Sa_v   ', &
-             'Sa_wspd', &
-             'Sa_tbot', &
-             'Sa_tskn', &
-             'Sa_shum'/)
-
-    do n = 1,size(flds)
-       fldname = trim(flds(n))
-       call addfld(fldListFr(compatm)%flds, trim(fldname))
-       call addfld(fldListTo(compocn)%flds, trim(fldname))
-    end do
-    deallocate(flds)
+    if (hafs_attr%atm_present .and. hafs_attr%ocn_present) then
+      allocate(S_flds(1))
+      S_flds = (/'So_t'/) ! sea_surface_temperature
+      do n = 1,size(S_flds)
+         fldname = trim(S_flds(n))
+         call addfld(fldListFr(compocn)%flds, trim(fldname))
+         call addfld(fldListTo(compatm)%flds, trim(fldname))
+      end do
+      deallocate(S_flds)
+    end if
 
     ! ---------------------------------------------------------------------
-    ! to ocn: zonal and meridional surface stress from atm
+    ! to atm: surface roughness length
     ! ---------------------------------------------------------------------
-    allocate(suffix(2))
-    suffix = (/'taux', 'tauy'/)
-
-    do n = 1,size(suffix)
-       call addfld(fldListFr(compatm)%flds , 'Faxa_'//trim(suffix(n)))
-       call addfld(fldListTo(compocn)%flds , 'Foxx_'//trim(suffix(n)))
-    end do
-    deallocate(suffix)
+    if (hafs_attr%atm_present .and. hafs_attr%wav_present) then
+      allocate(S_flds(1))
+      S_flds = (/'Sw_zo'/) ! wave_z0_roughness_length
+      do n = 1,size(S_flds)
+         fldname = trim(S_flds(n))
+         call addfld(fldListFr(compwav)%flds, trim(fldname))
+         call addfld(fldListTo(compatm)%flds, trim(fldname))
+      end do
+      deallocate(S_flds)
+    end if
 
     !=====================================================================
-    ! FIELDS TO ICE (compice)
+    ! FIELDS TO OCEAN (compocn)
```

Comment:
Is this OCEAN component HYCOM or dataocn?

Reply:
It is same for both data ocean and HYCOM.

Reply:
Just to confirm, so if you specify two components (HYCOM/data ocn) for ocean, med will send atm fields to both component? I thought data ocn only provides fields to other component, it does not receive any fields from other comp, is it not the case?

Reply:
No. I mean both data ocean and HYCOm configurations uses same set of fields / interface. So, there is no any distinction in the CMEPs layer about the coupling with the data ocean. When you use data ocean it does use import state and does not accept any field. Of course there are some cases that data ocean also receive information to do some calculation in their side but that ocean use in here not one of them.

Reply:
Thanks for the explanation.